src/core/utils.js

```diff
@@ -243,21 +243,11 @@ class Utils extends UtilsCore {
 	 *
 	 */
 	static isArray(array) {
-		switch (array.constructor) {
-			case Int8Array:
-			case Uint8Array:
-			case Uint8ClampedArray:
-			case Int16Array:
-			case Uint16Array:
-			case Int32Array:
-			case Uint32Array:
-			case Float32Array:
-			case Float64Array:
-			case Array:
-				return true;
-			default:
-				return false;
+		if((array.length || array.length === 0) && (array.constructor !== String)) {
```

Comment:
I am not sure I totally get what you are trying to say here. Can you elaborate the part where you say "that allows them to add their array type"?